PowerEditor/src/NppDarkMode.cpp

```diff
@@ -2852,4 +2854,69 @@ namespace NppDarkMode
 		}
 		return static_cast<INT_PTR>(NppDarkMode::onCtlColor(hdc));
 	}
+
+	struct HLSColour
+	{
+		WORD _Hue;
+		WORD _Lightness;
+		WORD _Saturation;
+
+		COLORREF toRGB() const { return ColorHLSToRGB(_Hue, _Lightness, _Saturation); }
+	};
+
+	using IndividualTabColours = std::array<HLSColour, 5>;
+
+	static constexpr IndividualTabColours individualTabHuesFor_Dark      { { HLSColour{0, 40, 60}, HLSColour{70, 40, 60}, HLSColour{144, 40, 60}, HLSColour{13, 40, 60}, HLSColour{195, 40, 60} } };
+	static constexpr IndividualTabColours individualTabHuesFor_DarkRed   { { HLSColour{0, 60, 60}, HLSColour{70, 50, 60}, HLSColour{144, 50, 60}, HLSColour{13, 50, 60}, HLSColour{195, 50, 60} } };
+	static constexpr IndividualTabColours individualTabHuesFor_DarkGreen { { HLSColour{0, 50, 60}, HLSColour{70, 50, 60}, HLSColour{144, 50, 60}, HLSColour{13, 50, 60}, HLSColour{195, 50, 60} } };
+	static constexpr IndividualTabColours individualTabHuesFor_DarkBlue  { { HLSColour{0, 50, 60}, HLSColour{70, 50, 60}, HLSColour{144, 50, 60}, HLSColour{13, 50, 60}, HLSColour{195, 50, 60} } };
+	static constexpr IndividualTabColours individualTabHuesFor_DarkPurple{ { HLSColour{0, 50, 60}, HLSColour{70, 50, 60}, HLSColour{144, 50, 60}, HLSColour{13, 50, 60}, HLSColour{195, 60, 60} } };
+	static constexpr IndividualTabColours individualTabHuesFor_DarkCyna  { { HLSColour{0, 60, 60}, HLSColour{70, 60, 60}, HLSColour{144, 70, 60}, HLSColour{13, 60, 60}, HLSColour{195, 60, 60} } };
+	static constexpr IndividualTabColours individualTabHuesFor_DarkOlive { { HLSColour{0, 60, 60}, HLSColour{70, 60, 60}, HLSColour{144, 60, 60}, HLSColour{13, 60, 60}, HLSColour{195, 60, 60} } };
+
+	static const IndividualTabColours individualTabHues              { { HLSColour{0, 210, 150}, HLSColour{70, 210, 150}, HLSColour{144, 210, 150}, HLSColour{13, 210, 150}, HLSColour{195, 210, 150} } };
```

Comment:
I see that `individualTabHues` (light theme) and all other 7 `IndividualTabColours individualTabHuesFor_DarkXXX` arrays contains different 5 colour values.

It will be a major problem in the future if we want to make these 5 colours customizable.
Is it possible to have a function in which we pass a RGB value, and choice (light, dark, darkRed, darkBlue...) then the function in question return the RGB we need?

Reply:
Notice that this is just implementation detail and what we are exposing is RGB `COLORREF getIndividualTabColour(...)`
So if in the future we want to add colour customization we only have to add a case for `if(g_colorToneChoice == customizedTone)`.

Reply:
>Is it possible to have a function in which we pass a RGB value, and choice (light, dark, darkRed, darkBlue...) then the function in question return the RGB we need?

And yes we can transform RGB into just a Hue and modify it according to theming rules back to RGB.

Reply:
@spaxio
> and modify it according to theming rules back to RGB.
 
When you say "modify it" you mean add it manually to other 7 vector right?
But if the customized color is completely dynamic added by user as we does in Style Configurator?
![image](https://user-images.githubusercontent.com/90293/189533296-d0762269-cb45-4bcb-af30-ee8236768ca9.png)